Modules/IO/ImageBase/include/itkImageFileWriter.h

```diff
@@ -240,58 +241,26 @@ class ITK_TEMPLATE_EXPORT ImageFileWriter : public ProcessObject
 /** Convenience function for writing an image. */
 template <typename TImage>
 ITK_TEMPLATE_EXPORT void
-WriteImage(const TImage * image, const char * filename, bool compress = false)
+WriteImage(TImage image, const std::string & filename, bool compress = false)
```

Comment:
Maybe rename `TImage` to `TImagePointer`? (Usually `TImage` itself is not a pointer type).

Reply:
Yes, I have that locally now :)

I am also experimenting with the following signature:
```
WriteImage(TImagePointer &image, const std::string & filename, bool compress = false)
```

So that if TImagePointer is a `SmartPointer` then a copy will not be made.

Reply:
> that if TImagePointer is a SmartPointer then a copy will not be made.

You might consider `TImagePointer &&`, if you really want to. I think the single ampersand version won't support passing a temporary pointer (rvalue) returned by a function.

Reply:
> > that if TImagePointer is a SmartPointer then a copy will not be made.
> 
> You might consider `TImagePointer &&`, if you really want to. I think the single ampersand version won't support passing a temporary pointer (rvalue) returned by a function.

and then I need to remove the the reference...

Reply:
Nitpick: This PR looks like a style improvement to me, so "STYLE", rather than "ENH". So I would suggest adjusting the subject line of the commit message

Reply:
The intent of the original code is still here so not much of an usage change, however this PR now adds substantial new meta-programming which should be considered and enhancement.

My guideline for STYLE vs ENH: When looking for a bug or a change in behavior can the commits with STYLE/DOC should be able to be skipped as no change in logic should occur.